amount.go

```diff
@@ -75,6 +76,24 @@ func NewAmount(n, currencyCode string) (Amount, error) {
 	return Amount{number, currencyCode}, nil
 }
 
+// NewAmountFromBigInt creates a new Amount from a big integer and a currency code.
+func NewAmountFromBigInt(amt *big.Int, currencyCode string) (Amount, error) {
+	if amt == nil {
+		return Amount{}, InvalidNumberError{"NewAmountFromBigInt", fmt.Sprint(amt)}
+	}
+	if currencyCode == "" || !IsValid(currencyCode) {
+		return Amount{}, InvalidCurrencyCodeError{"NewAmountFromBigInt", currencyCode}
+	}
+	d, _ := GetDigits(currencyCode)
+
+	return Amount{apd.NewWithBigInt(amt, -int32(d)), currencyCode}, nil
+}
+
+// NewAmount creates a new Amount from an int64 and a currency code.
+func NewAmountFromInt64(amt int64, currencyCode string) (Amount, error) {
+	return NewAmountFromBigInt(big.NewInt(amt), currencyCode)
```

Comment:
I like this, it matches what apd.New() does under the hood (wrap the int64 with a big.Int), and reduces boilerplate.